tf-psa-crypto/core/psa_util.c

```diff
@@ -365,7 +365,7 @@ int mbedtls_psa_get_random(void *p_rng,
 
 #endif /* MBEDTLS_PSA_CRYPTO_CLIENT */
 
-#if defined(MBEDTLS_PSA_UTIL_HAVE_ECDSA)
```

Comment:
Missing: removing the definition of `MBEDTLS_PSA_UTIL_HAVE_ECDSA`. Having this unused definition [created some confusion](https://github.com/Mbed-TLS/mbedtls/pull/9313#pullrequestreview-2230644518).

Reply:
@eleuzi01 Can you create a follow-up PR doing that and referencing this comment? Thanks!

Also, please check if there are other similar issues with other macros that have been replaced recently.

Reply:
There are similar issues because this was planned to be done in #9396 https://github.com/Mbed-TLS/mbedtls/pull/9384#issuecomment-2233434715
Would you still like me to do a follow-up PR for `MBEDTLS_PSA_UTIL_HAVE_ECDSA`?

Reply:
Ah, good point, wasn't aware of that ticket, thanks for pointing it out. I guess it's OK then, as long as #9396 is executed soon. I assume the plan is to do it as soon as #9337, #9252 and #9334 are completed? Since all of them seem to be in progress already, that shouldn't be too long hopefully.

(In retrospect I guess leaving that out till the end was perhaps not the best plan, but I fear changing plan midways may be causing more issues. So let's stick to the plan, but when doing #9396 we'll need to also double-check that the macros we're removing haven't started being used again by accident in the meantime.)

Reply:
Then, in such cases, can you please add a comment to explain why the dead code is staying? E.g.
```
/* This definition should no longer be used and will be removed soon.
 * https://github.com/Mbed-TLS/mbedtls/issues/9396 */
```